src/scipp/units/__init__.py

```diff
@@ -31,9 +30,171 @@
     :py:class:`scipp.Unit` to construct other units.
 """
 
+from contextlib import contextmanager
+from typing import Dict, Iterable, Tuple, Union
+
 from .._scipp.core.units import (angstrom, counts, default_unit, deg, dimensionless, kg,
                                  K, meV, m, one, rad, s, us, ns, mm)
 
+from .._scipp.core import (add_unit_alias as _add_unit_alias, clear_unit_aliases as
+                           _clear_unit_aliases)
+from ..core.cpp_classes import Unit, Variable
+
+
+class UnitAliases:
+    """Manager for unit aliases.
+
+    Aliases override how units are converted to and from strings.
+    The table is similar to a :class:`dict` and maps alias names to units.
+    But unlike a dict, no guarantees are made about the order of aliases or their
+    priority in string formatting.
+    And there may be only one alias for each unit at a time.
+
+    Attention
+    ---------
+    This class is a singleton and should never be instantiated by user code.
+    Instead, use it through :attr:`scipp.units.aliases`.
+    """
+
+    def __init__(self):
+        if any(map(lambda x: isinstance(x, UnitAliases), globals().values())):
+            raise RuntimeError("There can be only one instance of _Aliases")
+        self._aliases: Dict[str, Unit] = {}
+
+    def __setitem__(self, alias: str, unit: Union[str, Unit, Variable]):
+        """Define a new unit alias."""
+        multiplier, unit = _unit_and_multiplier(unit)
+        if unit in self.values():
+            raise ValueError(f"There already is an alias for unit '{unit!r}'")
```

Comment:
Is this check correct? Shouldn't it check *with* the multiplier? This looks like it would prevent defining an alias for different lattice spacings, e.g., in different directions.

Reply:
Right. Missed this when adding the separate multiplier.